pkg/controller/common/http.go

```diff
@@ -0,0 +1,33 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package common
+
+import (
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	internalProductRequestHeaderKey   = "x-elastic-product-origin"
+	internalProductRequestHeaderValue = "cloud"
+)
+
+// InternalProductHTTPHeaders is a slice that contains header that specifies which product is making the request.
+var InternalProductHTTPHeaders = []corev1.HTTPHeader{
+	{
+		Name:  internalProductRequestHeaderKey,
+		Value: internalProductRequestHeaderValue,
+	},
+}
+
+// SetInternalProductRequestHeader sets header that specifies which internal product is making the request.
+func SetInternalProductRequestHeader(req *http.Request) {
```

Comment:
I did not realized that this function is only called once, by the ES client, I think we could inline it.
Also I would consider having a `nil` parameter a programming error.
I was thinking about something along those lines in `base.go`:

```golang
	request, err := http.NewRequest(method, stringsutil.Concat(c.Endpoint, pathWithQuery), body) //nolint:noctx
	if err != nil {
		return err
	}

	// Sets headers allowing ES to distinguish between deprecated APIs used internally and by the user
	if request.Header == nil {
		request.Header = make(http.Header)
	}
	request.Header.Set(common.InternalProductRequestHeaderKey, common.InternalProductRequestHeaderValue)
```

_(Assuming `internalProductRequestHeaderKey` and `internalProductRequestHeaderValue` are public const)_